### ISSUE TITLE
Bad State: Stream has already been listened to

### ISSUE THREAD

**Issue by tomassasovsky** [rknell/alfred#72]
I'm getting this exception while reading req.body twice, once in the middleware, and once in the actual function.
the code i'm using:
```dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:alfred/alfred.dart';

void main() async {
  final app = Alfred();

  app.get('/example', (HttpRequest req, HttpResponse res) async {
    final body = await req.body;

    final jsonBody = jsonEncode(body.toString());
    return jsonBody;
  }, middleware: [
    (HttpRequest req, HttpResponse res) async {
      final body = await req.body;

      if (body is Uint8List) {
        throw AlfredException(401, {'message': 'authentication failed'});
      }
    }
  ]);

  await app.listen(3000, '127.0.0.1');
}
```

**Comment by rknell** [comment]
Hey @tomassasovsky 

Its being caused by the body being parsed twice. I'm working on a fix now that will just return a cached value if you try and listen twice. Should have it up soon.

**Comment by tomassasovsky** [comment]
Awesome, thanks!

**Comment by rknell** [comment]
All fixed, please see v0.1.5+2
